numba/core/typing/templates.py

```diff
@@ -517,7 +517,20 @@ def generic(self, args, kws):
             inline_worker = InlineWorker(tyctx, tgctx, fcomp.locals,
                                          compiler_inst, flags, None,)
 
-            ir = inline_worker.run_untyped_passes(disp_type.dispatcher.py_func)
+            # If the inlinee contains something to trigger literal arg dispatch
+            # then the pipeline will unconditionally fail, but the
+            # ForceLiteralArg instance that causes the failure will not have
+            # been through the path that adds `fold_arguments` to the exception,
+            # which means there's nothing to call if this exception propagates
+            # up to the dispatcher. As a result, it's best to pretend it didn't
+            # happen and continue to `resolve` as that will trigger a
+            # ForceLiteralArg exception instance that is appropriately formed
+            # for consumption in the dispatcher.
+            try:
+                ir = inline_worker.run_untyped_passes(
+                    disp_type.dispatcher.py_func)
+            except errors.ForceLiteralArg:
+                pass
             resolve = disp_type.dispatcher.get_call_template
             template, pysig, folded_args, kws = resolve(new_args, kws)
```

Comment:
```suggestion
            except errors.ForceLiteralArg:
                resolve = disp_type.dispatcher.get_call_template
                template, pysig, folded_args, kws = resolve(new_args, kws)
                raise errors.CompilerError("unreachable")
                 
            resolve = disp_type.dispatcher.get_call_template
            template, pysig, folded_args, kws = resolve(new_args, kws)
```

Reply:
following more out of band discussion, going to try swapping the order of the resolve and the pipieline call

Reply:
done in https://github.com/numba/numba/pull/5889/commits/c0e231b3c93136432067125b8cc600739c5ae692